tests/Doctrine/Tests/ORM/Functional/Ticket/DDC7561Test.php

```
@@ -0,0 +1,282 @@
+     */
+    public function __construct(int $id, string $amount, string $currencyIsoCode)
+    {
+        parent::__construct($id);
+        $this->amount = $amount;
+        $this->currencyIsoCode = $currencyIsoCode;
+        $this->DDC7561ConfirmationStatus = new DDC7561ConfirmationStatus('pending');
+    }
+
+    /**
+     * @return string
+     */
+    public function getAmount(): string
+    {
+        return $this->amount;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCurrencyIsoCode(): string
+    {
+        return $this->currencyIsoCode;
+    }
+
+    /**
+     * @return DDC7561ConfirmationStatus
+     */
+    public function getDDC7561ConfirmationStatus(): DDC7561ConfirmationStatus
+    {
+        return $this->DDC7561ConfirmationStatus;
+    }
+}
+
+class DDC7561DDC7561ConfirmationStatusType extends Type
+{
+    /**
+     * @param mixed            $value
+     * @param AbstractPlatform $platform
+     *
+     * @return string
+     *
+     * @throws ConversionException
+     */
+    public function convertToDatabaseValue($value, AbstractPlatform $platform): string
+    {
+        if (!$value instanceof DDC7561ConfirmationStatus) {
+            throw new ConversionException(\sprintf(
+                'Value must be instance of "%s", instance "%s" given',
+                DDC7561ConfirmationStatus::class,
+                \is_object($value) ? \get_class($value) : \gettype($value)
+            ));
+        }
+
+        /* @var DDC7561ConfirmationStatus $value */
+        return $value->getValue();
+    }
+
+    /**
+     * @param mixed            $value
+     * @param AbstractPlatform $platform
+     *
+     * @return DDC7561ConfirmationStatus
+     *
+     * @throws ConversionException
+     */
+    public function convertToPHPValue($value, AbstractPlatform $platform): DDC7561ConfirmationStatus
+    {
+        try {
+            return new DDC7561ConfirmationStatus($value);
+        } catch (\Throwable $e) {
```

Let's drop this try/catch completely - let's just let the exception bubble up.